hlv/hlv.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2023-Present Couchbase, Inc.
+
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+*/
+
+package hlv
+
+import (
+	"fmt"
+
+	"github.com/couchbase/goxdcr/base"
+)
+
+type ConflictDetectionResult uint32
+
+const (
+	Win      ConflictDetectionResult = iota
+	Lose     ConflictDetectionResult = iota
+	Conflict ConflictDetectionResult = iota
+	Equal    ConflictDetectionResult = iota
```

Comment:
Did we define "Equal" in the CCR document? 
In the document we talked about:
1. Dominate (Win)
2. Does not Dominate  (Lose)
3. Conflict

In keeping with the current scheme of conflict resolution, if it "Dominates", we send it. Otherwise, we don't send it (lose). There's no need for equal.

Reply:
In section: https://docs.google.com/document/d/1KTGXrRnxWMaRI6jwI0-0hoSmBHe3S6IEL3QvprhN_Lk/edit#heading=h.x50fvn11ltma 
HLV(A) == HLV(B) if
CV.source(A) == CV.source(B) and CV.Version(A) == CV.Version(B), or
They have identical non-empty MVs.

Reply:
Thanks